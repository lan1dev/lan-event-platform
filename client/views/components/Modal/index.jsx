import styled from 'styled-components';

/* eslint-disable react/destructuring-assignment */
const Modal = styled.div`
  background: ${props => (props.background ? props.background : 'white')};
  padding: 50px;
  margin: 50px;
`;
/* eslint-enable react/destructuring-assignment */

export default Modal;
