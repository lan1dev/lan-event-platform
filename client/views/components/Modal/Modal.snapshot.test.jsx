import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import Modal from './index';

describe('Modal', () => {
  let props;
  let wrapper;

  const Wrapper = () => {
    if (!wrapper) {
      wrapper = shallow(<Modal {...props}>{props.children}</Modal>);
    }
    return wrapper;
  };

  // Reset Component and Props before each test.
  beforeEach(() => {
    props = {};
    wrapper = undefined;
  });

  describe('without props', () => {
    it('Should match snapshot', () => {
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });

  describe('with background props', () => {
    it('Should match snapshot', () => {
      props = {
        background: 'gray'
      };
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });
});
