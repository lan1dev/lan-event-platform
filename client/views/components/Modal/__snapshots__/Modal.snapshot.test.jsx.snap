// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal with background props Should match snapshot 1`] = `
.c0 {
  background: gray;
  padding: 50px;
  margin: 50px;
}

<div
  className="c0"
/>
`;

exports[`Modal without props Should match snapshot 1`] = `
.c0 {
  background: white;
  padding: 50px;
  margin: 50px;
}

<div
  className="c0"
/>
`;
